components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -212,18 +257,36 @@ def run_clp(
 
     # Compute the total amount of data compressed
     last_archive_stats = None
-    total_uncompressed_size = 0
-    total_compressed_size = 0
-    while True:
+    last_line_decoded = False
+    worker_output = {
+        "total_uncompressed_size": 0,
+        "total_compressed_size": 0,
+    }
+
+    while not last_line_decoded:
         line = proc.stdout.readline()
-        if not line:
-            break
-        stats = json.loads(line.decode("ascii"))
-        if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
+        stats: Optional[Dict[str, Any]] = None
+        if line:
```

Comment:
If for some reason clp-s outputs an empty string in the middle of its archive stats output, wouldn't that mistake get hidden by `if line` (instead of `if line is None`)?

Reply:
any opinion on this? @kirkrodrigues

Reply:
Sorry, let's go with your suggestion.